addon-chrome/ChromePlacesProvider.js

```diff
@@ -161,8 +172,13 @@ module.exports = class ChromePlacesProvider {
    */
   static _getTopFrecentSites(histories) {
     return new Promise((resolve) => {
+      const searchResultRegex = new RegExp(SEARCH_RESULT_REGEX);
+      const localhostRegex = new RegExp(LOCALHOST_REGEX);
+      const browserResourceRegex = new RegExp(BROWSER_RESOURCE_REGEX);
+      const filterRegex = new RegExp(searchResultRegex.source + "|" + localhostRegex.source + "|" + browserResourceRegex.source);
```

Comment:
on regex bugs: http://stackstatus.net/post/147710624694/outage-postmortem-july-20-2016
and relevant discussions: https://news.ycombinator.com/item?id=12131909

;-)